### PR TITLE
S.sort as in R.sort

### PR DESCRIPTION
**Pull request by sadasant**
For: https://github.com/sanctuary-js/sanctuary/issues/353

Please review \o/

### CODE REVIEW

**Comment by davidchambers** [comment]
I left a comment in #353. Let's start with `sortBy` before deciding how `sort` should work.

File: index.js [hunk 1]
```diff
@@ -4030,6 +4030,25 @@
...
+  //. negative number if the first value is smaller, a positive number if
+  //. it's larger, and zero if they are equal. Please note that this is a
+  //. **copy** of the list. It does not modify the original.
+  //.
+  //. ```javascript
+  //. > S.sort(S.sub, [4, 2, 7, 5])
+  //. [2, 4, 5, 7]
+  //. ```
+  function sort(f, a) {
+    return a.sort(f);
```
Comment (davidchambers) [inline index.js]: This mutates `a`. :scream:

Comment (Avaq) [inline index.js]: :scream_cat:

Comment (sadasant) [inline index.js]: It's alright! A new version is coming soon 👀💧

Comment (sadasant) [inline index.js]: I have a comment here: https://github.com/sanctuary-js/sanctuary/issues/353#issuecomment-288889085


So, what about `def('sortBy', {f: [Z.Applicative, Z.Foldable, Z.Monoid]}, [Fn(a, Fn(a, Ordering)), f(a), f(a)], sortBy);` ?

Then we could use something like this: `sortBy cmp = foldr (insertBy cmp) []`, or should we rather implement the better/harder merge sort that's over here? http://hackage.haskell.org/package/base-4.9.1.0/docs/src/Data.OldList.html#sortBy

Little question: Do we have foldr in sanctuary-js?

Comment (sadasant) [inline index.js]: sortBy for review: https://github.com/sanctuary-js/sanctuary/pull/365

**Comment by davidchambers** [comment]
Let's add `sortBy` (#365) then revisit `sort`. :)

**Pull request closed without merging**